native/src/routes/settings/components/Settings.js

```diff
@@ -56,135 +50,97 @@ const SectionHeader = styled.Text`
   color: ${props => props.theme.colors.textColor};
 `
 
-export default class Settings extends React.Component<PropsType, StateType> {
-  appSettings: AppSettings
-  unsubscribeNavigationListener: () => void
-
-  constructor(props: PropsType) {
-    super(props)
+const appSettings = new AppSettings()
 
-    this.state = { settingsLoaded: false, settings: defaultSettings }
-    this.appSettings = new AppSettings()
+const Settings = ({ navigation, t, languageCode, cityCode, theme }: PropsType) => {
+  const [settings, setSettings] = useState<SettingsType | null>(null)
 
-    this.loadSettings()
+  useEffect(() => {
+    loadSettings()
+  }, [])
 
-    this.unsubscribeNavigationListener = this.props.navigation.addListener('focus', () => {
-      this.loadSettings()
+  useEffect(() => {
+    // Reload settings if navigating back from another route
+    return navigation.addListener('focus', () => {
+      loadSettings()
     })
-  }
+  }, [navigation])
```

Comment:
No need to remove the listener on unmount?
Any reason why the [useFocusEffect](https://reactnavigation.org/docs/function-after-focusing-screen/#triggering-an-action-with-the-usefocuseffect-hook) hook doesn't work here?

Reply:
It is removed on unmount, `navigation.addListener` directly returns the function to remove the listener as mentioned here: https://reactnavigation.org/docs/navigation-events/#navigationaddlistener
`useFocusEffect` seems fine, I just didn't look into refactoring this even more and also did not know about this one. Will try it :+1:

Reply:
Works nicely, thanks!